tests/test_0692_fsspec_writing.py

```diff
@@ -0,0 +1,125 @@
+        f["tree"] = {"x": np.array([1, 2, 3])}
+
+    with uproot.open(uri) as f:
+        assert f["tree"]["x"].array().tolist() == [1, 2, 3]
+
+
+def test_fsspec_writing_http(server):
+    pytest.importorskip("aiohttp")
+
+    uri = f"{server}/file.root"
+
+    with pytest.raises(NotImplementedError):
+        # TODO: review this when fsspec supports writing to http
+        with uproot.recreate(uri) as f:
+            f["tree"] = {"x": np.array([1, 2, 3])}
+
+        with uproot.open(uri) as f:
+            assert f["tree"]["x"].array().tolist() == [1, 2, 3]
+
+
+@pytest.mark.parametrize(
+    "scheme",
+    [
+        "",
+        "file://",
+        "simplecache::file://",
+    ],
+)
+def test_fsspec_writing_local_update(tmp_path, scheme):
+    uri = scheme + os.path.join(tmp_path, "some", "path", "file.root")
+    with uproot.recreate(uri) as f:
+        f["tree1"] = {"x": np.array([1, 2, 3])}
+
+    with uproot.update(uri) as f:
+        f["tree2"] = {"y": np.array([4, 5, 6])}
+
+    # read data and compare
+    with uproot.open(uri) as f:
+        assert f["tree1"]["x"].array().tolist() == [1, 2, 3]
+        assert f["tree2"]["y"].array().tolist() == [4, 5, 6]
+
+
+@pytest.mark.parametrize(
+    "scheme",
+    [
+        "ssh://",
+        "simplecache::ssh://",
+    ],
+)
+def test_fsspec_writing_ssh(tmp_path, scheme):
+    pytest.importorskip("paramiko")
+    import paramiko
+    import getpass
+
+    user = getpass.getuser()
+    host = "localhost"
+    port = 22
+
+    # only test this if we can connect to the host (this will work in GitHub Actions)
+    try:
+        with paramiko.SSHClient() as client:
+            client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            client.connect(hostname=host, port=port, username=user)
+    except (
+        paramiko.ssh_exception.SSHException,
+        paramiko.ssh_exception.NoValidConnectionsError,
+    ) as e:
+        pytest.skip(f"ssh connection to host failed: {e}")
+
+    local_path = os.path.join(tmp_path, "file.root")
+    uri = f"{scheme}{user}@{host}:{port}{local_path}"
+
+    with uproot.recreate(uri) as f:
+        f["tree"] = {"x": np.array([1, 2, 3])}
+
+    with uproot.open(uri) as f:
+        assert f["tree"]["x"].array().tolist() == [1, 2, 3]
+
+
+@pytest.mark.parametrize(
+    "scheme",
+    [
+        "memory://",
+        "simplecache::memory://",
+    ],
+)
+def test_fsspec_writing_memory(tmp_path, scheme):
+    uri = f"{scheme}{tmp_path}/file.root"
+
+    # when https://github.com/fsspec/filesystem_spec/pull/1426 is merged this will work, we must remove the catch
+    with pytest.raises(ValueError):
+        with uproot.recreate(uri) as f:
+            f["tree"] = {"x": np.array([1, 2, 3])}
+
+        with uproot.open(uri) as f:
+            assert f["tree"]["x"].array().tolist() == [1, 2, 3]
+
+        if scheme == "simplecache::memory://":
+            raise ValueError("wait for next fsspec release and remove this")
```

Comment:
Doesn't this raise an error?

Reply:
The error it's going to be catched on the outer scope. The idea is that the `memory://` will currently raise a `ValueError` because of a bug in fsspec that has already been corrected but not pushed to pypi yet. Once this is pushed to pypi this will no longer raise a `ValueError` and the test will fail, prompting to just remove the catches. I later added the `simplecache::memory//` scheme which for some reason doesn't fail so I added this manual raise which should also be removed.